### ISSUE TITLE
All pages styles are being bundled and delivered

### ISSUE THREAD

**Issue by kasadawa** [nuxt/nuxt#26987]
### Environment

Nuxt: 3.11
Bundler: vite
Css Preprocessor: less

### Reproduction

Createa a brand new project, 2 layouts, several pages*.server.vue. All of the layouts / pages should have styles inside. After building the project, going to a single layout -> page, all of the styles of all pages /layouts are being delivered.

### Describe the bug

Using Nuxt 3 SSR mode, with all pages being .server.vue deliveres by default inline styles only. 

This issue is whenever we add:

`  vite: {
    build: {
      cssCodeSplit: false,
    },
    }
  ` 
  With this config and .server.vue, there is a single styles.css file delivered, which contains all of the styles withing pages/ pages*.server.vue. 
I also have /layouts folder, it also combines all of the styles from the layouts. 

So the styles.css file, contain a lot of unused and conflicting styles. Before, with nuxt 2/ webpack I didn't have those issues at all. I was using inline styles and only the needed ones were delivered.

There is also an issue with the inlined-styles on the child components, which are using typescript. https://github.com/nuxt/nuxt/issues/25923

**Comment by huang-julien** [comment]
https://vitejs.dev/config/build-options#build-csscodesplit Setting css code split to false force vite to not split css by chunk and merge thel into a single css file.

**Comment by kasadawa** [comment]
Hey @huang-julien , 
Thanks for answering. I'm fully aware of that. Its merging everything , no matter if its used or not / or being called or not. The root or the problem is that it combines all of the styles, insead of needed ones. I suppose that it cannot re-generate styles based on route. As well due to using *.server.vue its duplicating the styles - 1 time they are added as inline and one time as styles.css file.


The default behavour should be for each component / page there should be a style file , but in case of using .server.vue on pages, it just inlines the styles into the html. I wouldn't mind on going with that, but I saw other side effect with it, which I linked above.
